Checkout@v2 use `Git branch --show-current` which requires git 2.22.0

The v2 checkout uses `git branch --show-current` at https://github.com/actions/checkout/blob/master/src/git-command-manager.ts#L177

The `--show-current` was introduced in git [2.22.0](https://git-scm.com/docs/git-branch/2.22.0) but the checkout@v2 only requires [2.18.0](https://github.com/actions/checkout/blob/master/src/git-command-manager.ts#L11), when using checkout@v2 in a self-hosted runner, you will get an error about the `--show-current` option.
```
"H:\Program Files\Git\cmd\git.exe" branch --show-current
##[error]error: unknown option `show-current'
```

Seems like we need to bump the min required git version?

--- a/src/git-command-manager.ts
+++ b/src/git-command-manager.ts
@@ -170,12 +170,12 @@ class GitCommandManager {
   }
 
   async isDetached(): Promise<boolean> {
-    // Note, this implementation uses "branch --show-current" because
-    // "rev-parse --symbolic-full-name HEAD" can fail on a new repo
-    // with nothing checked out.
-
-    const output = await this.execGit(['branch', '--show-current'])
-    return output.stdout.trim() === ''
+    // Note, "branch --show-current" would be simpler but isn't available until Git 2.22
+    const output = await this.execGit(
+      ['rev-parse', '--symbolic-full-nane', 'HEAD'],
+      true
+    )
+    return !output.stdout.trim().startsWith('refs/heads/')
   }
 
   async lfsFetch(ref: string): Promise<void> {

--- a/src/git-command-manager.ts
+++ b/src/git-command-manager.ts
@@ -77,10 +77,12 @@ class GitCommandManager {
   async branchList(remote: boolean): Promise<string[]> {
     const result: string[] = []
 
-    // Note, this implementation uses "rev-parse --symbolic" because the output from
+    // Note, this implementation uses "rev-parse --symbolic-full-name" because the output from
     // "branch --list" is more difficult when in a detached HEAD state.
+    // Note, this implementation uses "rev-parse --symbolic-full-name" because there is a bug
+    // in Git 2.18 that causes "rev-parse --symbolic" to output symbolic full names.
 
-    const args = ['rev-parse', '--symbolic']
+    const args = ['rev-parse', '--symbolic-full-name']
     if (remote) {
       args.push('--remotes=origin')
     } else {
@@ -92,6 +94,12 @@ class GitCommandManager {
     for (let branch of output.stdout.trim().split('\n')) {
       branch = branch.trim()
       if (branch) {
+        if (branch.startsWith('refs/heads/')) {
+          branch = branch.substr('refs/heads/'.length)
+        } else if (branch.startsWith('refs/remotes/')) {
+          branch = branch.substr('refs/remotes/'.length)
+        }
+
         result.push(branch)
       }
     }

--- a/src/git-command-manager.ts
+++ b/src/git-command-manager.ts
@@ -172,7 +172,7 @@ class GitCommandManager {
   async isDetached(): Promise<boolean> {
     // Note, "branch --show-current" would be simpler but isn't available until Git 2.22
     const output = await this.execGit(
-      ['rev-parse', '--symbolic-full-nane', 'HEAD'],
+      ['rev-parse', '--symbolic-full-name', 'HEAD'],
       true
     )
     return !output.stdout.trim().startsWith('refs/heads/')

--- a/src/git-command-manager.ts
+++ b/src/git-command-manager.ts
@@ -172,7 +172,7 @@ class GitCommandManager {
   async isDetached(): Promise<boolean> {
     // Note, "branch --show-current" would be simpler but isn't available until Git 2.22
     const output = await this.execGit(
-      ['rev-parse', '--symbolic-full-name', 'HEAD'],
+      ['rev-parse', '--symbolic-full-name', '--verify', '--quiet', 'HEAD'],
       true
     )
     return !output.stdout.trim().startsWith('refs/heads/')

